File: tower_of_hanoi.cpp
/*
Problem Statement:
Given n rings, move from tower A to tower C using B as auxillary. 
Output: the number of steps
*/
#include<iostream>
#include<stdlib.h>
using namespace std;
enum class TOWER{
    A,
    B,
    C
};

class TowerOfHonai{
    public:
    static int move(int n, TOWER A, TOWER C, TOWER B){
        int steps;
        if(n==0 || n==1){
            return n;
        }
        steps = move(n-1,A,B,C);
        ++steps;
        steps += move(n-1,B,C,A);
        return steps;

    }
};

int main(int argc, char *argv[]){
    if(argc != 2){
        cerr<<"Insufficient number of arguments";
        return -1;
    } 
    int n = atoi(argv[1]);
    int steps = TowerOfHonai::move(n,TOWER::A,TOWER::C,TOWER::B);
    std::cout<<"number of steps to move "<<n<<" rings from tower A to C is"<<steps;
    return 0;

}

File: tower_of_hanoi.py

'''
Move n towers from A to B with the help of C
'''
class TowerOfHanoi:
    def __init__(self, n):
        self.n = n
    
    def moveRings(self, n, A,B,C):
        if(n == 0 or n == 1):
            return n
        steps = self.moveRings(n-1, A, C, B)
        steps += 1
        steps += self.moveRings(n-1, C, B, A)
        return steps
    def moveTowerOfHanoi(self):
        return self.moveRings(n, "A", "B", "C")
    
if(__name__ == "__main__"):
     n = 5
     th = TowerOfHanoi(n)
     steps = th.moveTowerOfHanoi()
     print("number of steps to move "+ str(n) + " rings is "+ str(steps))

    
    



            